### ISSUE TITLE
[Bug]: Booking feature bugs

### ISSUE THREAD

**Issue by jurrejansen** [Shelf-nu/shelf.nu#770]
### What happened?

Multiple bugs

### What is the expected behaviour?

- Assets that are marked as "Unavailable for bookings" are hidden in the Manage assets modal when configuring a booking. However, you can easily make them appear by using the pagination. This is the case for Administrator and Self-service users.
https://github.com/Shelf-nu/shelf.nu/assets/76112272/9b7da976-ef3b-424a-bf72-89417955c268

- 2 Draft bookings with overlapping booking periods, in 1 booking I add assets but do not Reserve and keep as Draft. When I go to the other booking and try to add these assets they're already marked as "Already booked". This goes against the concept of a Draft imo. These assets are not yet booked if the booking is still a Draft.
https://github.com/Shelf-nu/shelf.nu/assets/76112272/76bd327c-27a3-440b-b32f-f4053612b374

- As a Self-service user I am able to Reserve a booking, obviously, but after I am still able to open the Manage assets modal and add whatever I like to my Reserved booking. This includes asset that are In custody etc. I think we should not give a Self-service user the option to open the Manage assets modal after the booking reached Reserved status.
https://github.com/Shelf-nu/shelf.nu/assets/76112272/f25d7a94-4769-4f76-80e0-ec801a11061f

### Version

Hosted (app.shelf.nu)

### What browsers are you seeing the problem on?

Chrome

**Comment by carlosvirreira** [comment]
Good finds @jurrejansen

**Comment by DonKoko** [comment]
> 2 Draft bookings with overlapping booking periods, in 1 booking I add assets but do not Reserve and keep as Draft. When I go to the other booking and try to add these assets they're already marked as "Already booked". This goes against the concept of a Draft imo. These assets are not yet booked if the booking is still a Draft.

I agree with you but I dont know why for some reason we were considering draft here as well. 

> As a Self-service user I am able to Reserve a booking, obviously, but after I am still able to open the Manage assets modal and add whatever I like to my Reserved booking. This includes asset that are In custody etc. I think we should not give a Self-service user the option to open the Manage assets modal after the booking reached Reserved status.

Agreed.

Both are fixed in: #781

**Comment by DonKoko** [comment]
@jurrejansen being deployed to live now. Will be online in a few. Please re-open the issue if some of those specific issues persist.

**Comment by jurrejansen** [comment]
@DonKoko 

- [ ] Administrator and Self-service users are able to add an 'In custody' asset to a booking and perform the Reserve action. For both user it should not be possible to perform the Reserve action.
https://github.com/Shelf-nu/shelf.nu/assets/76112272/51633b31-d7bd-4feb-8cad-aba8afc8eb23

- [ ] Self-service users are unable to use the Manage assets modal after booking reaches Reserved state. However we should change the tooltip that shows when hovering the button.

"Unfortunately you are unable to manage assets at this point. Cancel this booking and create another one if you need to make changes."

![Screenshot 2024-02-26 at 13 01 02](https://github.com/Shelf-nu/shelf.nu/assets/76112272/6e88f22a-75d9-4486-bb4f-ec57d209958a)

**Comment by DonKoko** [comment]
@jurrejansen i fixed those 2 things as we discussed. Will be deployed with #801

**Comment by jurrejansen** [comment]
@DonKoko re-opening this one because I found a bug

I invited myself as an Administrator to a Team workspace via the Owner account. 

When configuring a booking on the Owner account I am able to choose both the Owner user (myself) and the Administrator user as the Custodian

When configuring a booking on the Administrator account I don't see the Owner user when choosing a custodian. Instead I see the Administrator user (myself) twice for some reason.

https://github.com/Shelf-nu/shelf.nu/assets/76112272/0d3495b8-21b7-4650-b625-a81fec98a316
